### PR TITLE
[client/catapult] feat: catapult client ci

### PR DESCRIPTION
**Pull request by Wayonb**
## What is the current behavior?
Catapult client only does linting in its CI

## What's the issue?
Need to add build and testing

## How have you changed the behavior?
Add build and tests to the catapult client CI
1. docker args for caching volumes
2. mount the repo root instead of client source.  Code coverage tool requires access to the codecov.yaml file
3. update the inner build scripts to take source and output path

## How was this change tested?
Yes.

### CODE REVIEW

File: client/catapult/scripts/ci/build.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+else
+	output_path="${git_root_directory}/output/binaries"
+	mkdir -p "${output_path}"
+	python3 "${scripts_directory}/runDockerBuildInnerBuild.py" \
+		--compiler-configuration "${scripts_directory}/configurations/gcc-11.yaml" \
+		--build-configuration "${build_configuration}" \
+		--source-path "${git_root_directory}/client/catapult" \
+		--out-dir "${output_path}"
+fi
+
```
Comment (Jaguar0625) [inline client/catapult/scripts/ci/build.sh:26]: trailing space

File: jenkins/docker/cpp.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	| tee /etc/apt/sources.list.d/mongodb-org-5.0.list \
+	&& apt-get update \
+	&& apt-get install -y mongodb-org
+
+# add ubuntu user (used by jenkins)
+RUN useradd --uid 1000 -ms /bin/bash ubuntu
+
+# Create the MongoDB data directory
+RUN mkdir -p /data/db \
+	&& chown -R ubuntu:ubuntu /data
```
Comment (Jaguar0625) [inline jenkins/docker/cpp.Dockerfile:20]: unsure if needed, but missing WORKDIR like other Dockerfiles

File: client/catapult/scripts/ci/build.sh [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+set -ex
+
+git_root_directory="$(git rev-parse --show-toplevel)"
+scripts_directory="${git_root_directory}/jenkins/catapult"
+build_configuration="${scripts_directory}/configurations/tests-diagnostics.yaml"
+if [ -z "${JENKINS_HOME}" ]
+then
+	python3 "${scripts_directory}/runDockerBuild.py" \
+		--compiler-configuration "${scripts_directory}/configurations/gcc-10.yaml" \
```
[the remark this answers is not in the document]
Comment (Wayonb) [reply client/catapult/scripts/ci/build.sh:11]: Its  gcc 11 in Jenkins and 10 local.  
The test image for ubuntu is currently gcc 10 and needs to be updated.  If the build is done with gcc11, the test image is created with gcc 10 thus causing the test to fail.

I was going to do this in another PR after talking to @gimre-xymcity about it.

**Review by gimre-xymcity — APPROVED** [review]
jags commets aside, looks good

**Review by Jaguar0625 — APPROVED** [review]

**Pull request merged**